src/prototype_room_creepbuilder.js

```diff
@@ -48,6 +48,10 @@ Room.prototype.spawnCreateCreep = function(role, heal, level, squad, routing, ba
       routing: routing
     };
     var returnCode = spawn.createCreep(partConfig, name, memory);
+    var userName = Memory.username || Game.rooms[Memory.myRooms[0]].controller.owner || 'default';
+    let roleStat = Memory.stats[Memory.username].roles[role];
+    let previousAmount = roleStat ? roleStat.amount : 0;
+    brain.stats.add('', '.roles.' + role, previousAmount + 1);
```

Comment:
`  let userName = Memory.username || Game.rooms[Memory.myRooms[0]].controller.owner || 'default';
    // Cleanup memory		    // Cleanup memory
    for (let name in Memory.creeps) {		    for (let name in Memory.creeps) {
      if (!Game.creeps[name]) {		      if (!Game.creeps[name]) {
       let role = Memory.creeps[name].role;
       let roleStat = Memory.stats[userName].roles[role];
       let previousAmount = roleStat ? roleStat.amount : 0;
       let amount = previousAmount ? 0 : previousAmount - 1;
       brain.stats.add('', '.roles.' + role, amount);
 `
is almost identical in brain_memory.jsL44